File: programmers/p1_42748_k번째수_kit_sort_1.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int index = 0;
        for(int[] command : commands){
            int[] arraySliced = Arrays.copyOfRange(array, command[0]-1, command[1]);
            Arrays.sort(arraySliced);
            answer[index++] = arraySliced[command[2]-1];
        }
        return answer;
    }
}


/*
문제주소 : https://programmers.co.kr/learn/courses/30/lessons/42748?language=java
시간 : 측정안함


다른 사람 풀이 :
========================================================================================
for문 이용 차이

import java.util.Arrays;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        for(int i=0; i<commands.length; i++){
            int[] temp = Arrays.copyOfRange(array, commands[i][0]-1, commands[i][1]);
            Arrays.sort(temp);
            answer[i] = temp[commands[i][2]-1];
        }

        return answer;
    }
}
========================================================================================
기타
ArrayList <-> array 한 풀이
직접 array sort 작성한 풀이
========================================================================================

노트 :
- 자바 기본 자료형 다루는 함수명이 기억이 잘 안났음
- util 좀더 공부하면 좋을것 같음
- for문 index 필요할때 기본 for문이 더 편할수도

 */

File: programmers/p1_42748_k번째수_kit_sort_1.py
def solution(array, commands):
    answer = []
    for command in commands:
        slice_array = array[command[0]-1:command[1]]
        slice_array.sort()
        answer.append(slice_array[command[2]-1])
    return answer


"""
문제주소 : https://programmers.co.kr/learn/courses/30/lessons/42748?language=python3
시간 : 측정안함


다른 사람 풀이 :
========================================================================================
sorted 활용

def solution(array, commands):
    answer = []
    for com in commands:
        answer.append(sorted(array[com[0]-1:com[1]])[com[2]-1])
    return answer
========================================================================================
map에 람다 활용

def solution(array, commands):
    return list(map(lambda x:sorted(array[x[0]-1:x[1]])[x[2]-1], commands))
========================================================================================
배열 변수 세개에 할당

def solution(array, commands):
    result = []
    for command in commands:
        i, j, k = command
        result.append(sorted(array[i-1:j])[k-1])
    return result
========================================================================================

노트 :
파이썬 내장함수, 라이브러리 익숙해져야함

"""